### DISCUSSION TITLE
Help with remote dev container

### DISCUSSION THREAD

**Discussion by jalanh11640** [devcontainers/community#133] [Q&A] [answered]
I just joined a team that is using docker containers.  I've never used containers before.  I'm trying to figure out how to best incorporate vscode into our container setup/process.  The guy on our team that has setup our dev container provided a means to run the container, then attach locally in a terminal.  I've been trying to figure out how to use vscode to attach to the container.  I've found a process that works for me but it's cumbersome.

1) Using Remote SSH connect to our linux box
2) Using the developed script, start our dev container
3) In the remote explorer, select dev containers, then attach to the newly started container.
4) Open the associated container configuration
5) If the configuration is blank (we're still working on getting everything we need into the container, add remoteUser/extensions, etc
6) Close the remote connection and re-connect to the container again.

I played around some with a devcontainer.json file.  This seems to be a more streamlined way to go.. however, I ran into a number of problems.  The primary problem being that our script to run/connect to the container does a number of operations after the container has been started (ex 'docker exec adduser ...', 'docker cp <local dot file> <remote location>', etc).

Is there a way to configure devcontainer.json to run the 'start container' script we already have, then just connect to the container?  If that's not possible, is there a way to issue 'docker exec/cp/etc' commands after vscode has started the docker?  Whenever I would run with a devcontainer.json the process would fail because no user had been setup yet.

Thanks for any help.
Jeff

**Comment by samruddhikhandale** [comment, marked as answer]
Hi 👋 

Welcome to the Dev Container community.
Have you looked at https://containers.dev/implementors/json_reference/#lifecycle-scripts document? 

You can configure the `devcontainer.json` file to run a lifecycle script after the container has started. You can use the `postCreateCommand` or `postStartCommand` properties in the `devcontainer.json` file to run commands after the container has been created or started, respectively.

Please note that the script file should be in a location that is accessible to the container. You can add the script file to your project and use a relative path to it in the devcontainer.json file.

Also, keep in mind that these commands run inside the container, not on the host machine. If you need to run commands on the host machine, you might need to adjust your setup or script accordingly.

**Reply by jalanh11640** [reply]
Awesome, I'll look into those.  Thanks for the information!
